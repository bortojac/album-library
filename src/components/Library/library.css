.library {
  display: flex;
  flex: 1;
  flex-flow: column;
  align-items: center;
  min-height: 40%;
  width: 100%;
}

.library__h1 {
  font-size: 1.5rem;
  width: 100%;
  align-self: flex-start;
}

.library__button {
  padding: .5rem;
  margin: 1rem 0;
  height: 4rem;
  border: solid .2rem #000;
  cursor: pointer;
}

@media screen and (min-width: 641px) {
  .library__h1 {
    width: 50%;
  }
  .library {
    width: 50%;
  }
}