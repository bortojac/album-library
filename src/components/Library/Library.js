import React, { Component } from 'react';
import AlbumList from '../AlbumList';
import './library.css';

class Library extends Component {
  constructor(props) {
    super(props);
    this.state = {
      load: 3,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState((prevState) => ({
      load: prevState.load + 3
    }));
  }

  render() {
    return (
      <div className="library">
        <h1 className="library__h1">The Top Shelf</h1>
          <AlbumList searchFlag={false} load={this.state.load} />
        <button
          className="library__button"
          type="button"
          onClick={this.handleClick}>
            Load more
          </button>
      </div>
    );
  }
}

export default Library;