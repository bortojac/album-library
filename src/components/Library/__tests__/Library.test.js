import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Library from '../Library';

jest.mock('../../AlbumList');

// check render
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Library />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot
it('renders correctly', () => {
  const tree = renderer
    .create(<Library />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
