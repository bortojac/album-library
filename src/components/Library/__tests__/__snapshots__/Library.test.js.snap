// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="library"
>
  <h1
    className="library__h1"
  >
    The Top Shelf
  </h1>
  <button
    className="library__button"
    onClick={[Function]}
    type="button"
  >
    Load more
  </button>
</div>
`;
